modules/pipeline/providers/leaderworker/worker/type.go

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package worker
+
+import (
+	"github.com/google/uuid"
+)
+
+type ID string
+
+func NewID() ID              { return ID(uuid.New().String()) }
+func (id ID) String() string { return string(id) }
```

Comment:
should encapsulate a common uuid pkg

Reply:
en, add to `pkg/crypto/uuid` pkg, and add a new method of `UUIDWithDash`.

Reply:
in another pr after this merged.

Reply:
see: https://github.com/erda-project/erda/pull/4286

Reply:
adjusted